File: dsl-api/src/main/java/cbs/dsl/api/context/FinishContext.java
package cbs.dsl.api.context;

import lombok.Builder;

import java.util.Map;

@Builder(toBuilder = true)
public record FinishContext(
    String eventNumber,
    String performedBy,
    Map<String, Object> params,
    HelperEvaluator helperEvaluator) {

  public FinishContext put(String key, Object value) {
    params.put(key, value);
    return this;
  }

  public Object get(String key) {
    return params.get(key);
  }

  public Object helper(String key, Map<String, Object> values) {
    return helperEvaluator.evaluate(key, values);
  }

  public FinishContext copy() {
    return toBuilder().build();
  }
}

File: dsl-api/src/main/kotlin/cbs/dsl/api/context/FinishContext.kt
package cbs.dsl.api.context

open class FinishContext(
  eventCode: String,
  workflowExecutionId: Long,
  performedBy: String,
  dslVersion: String,
  eventParameters: Map<String, Any>,
) : EnrichmentContext(eventCode, workflowExecutionId, performedBy, dslVersion, eventParameters)
